python/cudf_polars/cudf_polars/experimental/repartition.py

```diff
@@ -59,9 +59,11 @@ def _(
         )
 
     key_name = get_key_name(ir)
-    stride = math.ceil(count_in / count_out)
+    n = count_in // count_out
+    remainder = count_in - count_out * n
+    offsets = [0, *itertools.accumulate(n + (i < remainder) for i in range(count_out))]
```

Comment:
I guess the intent of this partitioning scheme is to evenly distribute out the "extra" items so that we don't end up with one very imbalanced partition? If so, could we add a comment to that effect? A naive implementation of this would just be to scale the last offset to a larger stride to account for the remaining items and wouldn't require the fancier prefix sum approach.

Reply:
Yes, in the worst case shoving everything on the end can produce a partition that is asymptotically twice as large as the others, added a comment.